packages/mdc-theme/_functions.scss

```diff
@@ -51,3 +56,63 @@
     @return "light";
   }
 }
+
+// lighten() and darken() require values to be between 0% and 100%.
+@function mdc-theme-clamp-percentage_($percentage) {
+  @return max(0%, min(100%, $percentage));
+}
+
+// Generate light and dark variants of the given color, offset by approximately
+// the specified number of indexes within the color's tonal palette.
+@function mdc-theme-tonal-variants_($color, $num-indexes: 2) {
+  $luminance: mdc-theme-luminance($color);
+  $amount-1x: mdc-theme-clamp-percentage_($_mdc-theme-tonal-offset * $num-indexes);
+  $amount-2x: mdc-theme-clamp-percentage_($_mdc-theme-tonal-offset * $num-indexes * 2);
+
+  @if $luminance < .2 {
```

Comment:
Yeah, I totally misinterpreted this before and this seems like a clearer way to organize it.

Random thought (maybe a design question): is there a reason these thresholds are absolute rather than relative to the tonal offset?